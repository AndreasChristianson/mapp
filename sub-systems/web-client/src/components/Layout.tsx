import styled from 'styled-components'
import Log from './Log'

const FlexContainer = styled.div`
  display: flex;
  flex-direction: row;
  width: 100%;
  justify-content: space-evenly;
`

const Layout = () => (
  <FlexContainer>
    <Log event="uuid" showStats={true} />
    <Log event="log" />
  </FlexContainer>
)

export default Layout
