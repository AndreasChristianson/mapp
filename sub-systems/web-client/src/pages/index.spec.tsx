import renderer from 'react-test-renderer'
import Chance from 'chance'

import Index from './index'

import Layout from '../components/Layout'

jest.mock('../components/Layout')

describe('Index', () => {
  const chance = new Chance()
  let randomUserAgent: string
  beforeEach(() => {
    ;(Layout as jest.Mock).mockReturnValue(null)
    randomUserAgent = chance.sentence()
  })

  it('should match snapshot', () => {
    const tree = renderer.create(<Index userAgent="non-random user agent" />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  describe('getInitialProps', () => {
    let context: any
    describe('server side rendered', () => {
      beforeEach(() => {
        context = {
          req: {
            headers: {
              'user-agent': randomUserAgent,
            },
          },
        }
      })
      describe('user-agent header provided', () => {
        it('should extract the user agent', async () => {
          const props = await Index.getInitialProps!(context)

          expect(props.userAgent).toBe(randomUserAgent)
        })
      })
      describe('user-agent header not provided', () => {
        beforeEach(() => {
          delete context.req.headers['user-agent']
        })

        it('should extract the user agent', async () => {
          const props = await Index.getInitialProps!(context)

          expect(props.userAgent).toBe('')
        })
      })
    })

    describe('client side rendered', () => {
      beforeEach(() => {
        context = {}
        jest.spyOn(window.navigator, 'userAgent', 'get').mockReturnValue(randomUserAgent)
      })

      it('should extract the user agent', async () => {
        const props = await Index.getInitialProps!(context)

        expect(props.userAgent).toBe(randomUserAgent)
      })
    })
  })
})
