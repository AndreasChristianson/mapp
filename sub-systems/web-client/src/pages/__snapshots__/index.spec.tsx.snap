// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should match snapshot 1`] = `
Array [
  "stats component",
  <hr />,
  "room component",
]
`;
